src/gui/symbology/qgscategorizedsymbolrendererwidget.cpp

```diff
@@ -831,6 +831,9 @@ void QgsCategorizedSymbolRendererWidget::addCategories()
   {
     QgsCategoryList prevCats = mRenderer->categories();
     keepExistingColors = !prevCats.isEmpty();
+    QgsRandomColorRamp *randomColors = new QgsRandomColorRamp();
```

Comment:
leaky - you should be able to avoid the heap creation here entirely and just create on the stack instead...